### PR TITLE
feature/click-uri-updates

### PR DESCRIPTION
**Pull request by fivetran-joemarkiewicz**
## PR Overview
**This PR will address the following Issue/Feature:** Issue #35 and Issue #30

**This PR will result in the following new package version:** `v0.10.0`

This will be a breaking change do to the addition of the new `click_uri_type` field and changes to the underlying `click_uri` field.

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

> ## Breaking Changes
> - The `click_uri_type` field has been added to the below mentioned models. This field allows users to differentiate which click uri type (`text_ad` or `spotlight`) is being used to populate the results of the `click_uri` field. Please be aware this field only supports `text_ad` or `spotlight` click uri types. If you are interested in this package supporting more click uri ad types, please let us know in this [Feature Request](https://github.com/fivetran/dbt_linkedin_source/issues/70).
>   - `stg_linkedin_ads__creative_history`
>   - `linkedin_ads__creative_report`
>   - `linkedin_ads__url_report`
> 
> ## Bug Fixes (upstream dbt_linkedin_source change)
> - The `click_uri` field has been adjusted to populate the results following a coalesce on the `text_ad_landing_page`, `spotlight_landing_page`, or `click_uri` fields. For more details refer to [dbt_linkedin_source v0.10.0](https://github.com/fivetran/dbt_linkedin_source/releases/tag/v0.10.0) release notes.
>   - This change is in response to a [LinkedIn Ads API](https://learn.microsoft.com/en-us/linkedin/marketing/community-management/contentapi-migration-guide?view=li-lms-2024-05#adcreativesv2-api-creatives-api) and [Fivetran LinkedIn Ads connector update](https://fivetran.com/docs/connectors/applications/linkedin-ads/changelog#january2024) which moved `click_uri` data to either the `text_ad_landing_page` or `spotlight_landing_page` fields depending on the creative type.
> 
> ## Documentation Updates
> - The `click_uri` field documentation has been updated to reflect the updated state of the field.
> - Included a new `DECISIONLOG.md` entry to highlight why it's possible metrics don't add up across different grains.
> 
> ## Under the Hood
> - Updates to the `linkedin_creative_history_data` seed file to include the following new fields to ensure accurate data validation tests:
>   - `text_ad_landing_page`
>   - `spotlight_landing_page`

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [X] dbt run –full-refresh && dbt test
- [n/a] dbt run (if incremental models are present) && dbt test

Before marking this PR as "ready for review" the following have been applied:
- [X] The appropriate issue has been linked, tagged, and properly assigned
- [X] All necessary documentation and version upgrades have been applied
- [X] docs were regenerated (unless this PR does not include any code or yml updates)
- [x] BuildKite integration tests are passing
- [X] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:

Please see successful validation test results below:
image

In addition to the validations succeeding above, I was able to confirm the new `click_uri_type` field is populating as expected. See the results below using our seed data.
- linkedin_ads__creative_report you can see for relevant creatives that the `click_uri` and `click_uri_type` fields are populating as expected.
   - image
- linkedin_ads__url_report you can see for all records that the `click_uri` and `click_uri_type` fields are populating as expected. In this screenshot we can see that the `http://www.github.com` click_uri is showing a `null` uri type. This is expected because as we know from the seed data, the `http://www.github.com` click_uri data is populated from the now deprecated `click_uri` field from the source package. This should appropriately not have a click_uri_type. Only the now supported text_ad and spotlight types should populate.
   - image

### If you had to summarize this PR in an emoji, which would it be?
 
🔗

### CODE REVIEW

File: packages.yml [hunk 1]
```diff
@@ -1,3 +1,6 @@
 packages:
-  - package: fivetran/linkedin_source
-    version: [">=0.9.0", "<0.10.0"]
\ No newline at end of file
+  # - package: fivetran/linkedin_source
+  #   version: [">=0.10.0", "<0.11.0"]
+  - git: https://github.com/fivetran/dbt_linkedin_source.git
+    revision: feature/click-uri-deprecation
+    warn-unpinned: false
```
Comment (fivetran-joemarkiewicz) [inline packages.yml:6]: Will update before merge and release.

File: models/linkedin.yml [hunk 2]
```diff
@@ -389,9 +389,10 @@ models:
           CANCELED - Creative is permanently canceled.
 
       - name: click_uri
-        description: |
-          Required when creative type is TEXT_AD. Maximum character limit of 500 after URL encoding all special characters. 
-          Limit does not include the "https://" prefix.
+        description: The URL where the member should be redirected to, on clicking the text ad or spotlight ad.
+
+      - name: click_uri_type
+        description: Will either be `text_ad`, `spotlight`, or `null` to indicate the origin of the click_uri.
```
Comment (fivetran-reneeli) [inline models/linkedin.yml:395]: Same question as in the source


> I feel like null can mean the click_uri either originates from the old click_uri field or just another click uri ad type. Do you feel it's fine that this scenario isn't differentiated?

Comment (fivetran-reneeli) [inline models/linkedin.yml:395]: Same question as in the source


> I feel like null can mean the click_uri either originates from the old click_uri field or just another click uri ad type. Do you feel it's fine that this scenario isn't differentiated?

**Review by fivetran-reneeli — APPROVED** [review]
Looks good with just one question!

File: models/linkedin.yml (hunk 2 above)
Comment (fivetran-reneeli) [reply models/linkedin.yml:395]: Never mind, thanks for answering this in the source!

Comment (fivetran-joemarkiewicz) [reply models/linkedin.yml:395]: Thanks @fivetran-reneeli from the source discussion, this should be fine as is.

**Review by fivetran-avinash — APPROVED** [review]
@fivetran-joemarkiewicz PR approved, with standard reminder to update `packages.yml`

**Review by fivetran-catfritz — APPROVED** [review]
@fivetran-joemarkiewicz I was able to run the package updates and see that the `click_uri` coalesce is working properly, and the `click_uri_type` is populated correctly. I also tested the partitioning and `source_relation` logic, and that is compiling correctly. LGTM!

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,25 @@
+# dbt_linkedin v0.10.0
+[PR #38](https://github.com/fivetran/dbt_linkedin/pull/38) includes the following updates:
+
+## Breaking Changes
+- The `click_uri_type` field has been added to the below mentioned models. This field allows users to differentiate which click uri type (`text_ad` or `spotlight`) is being used to populate the results of the `click_uri` field. Please be aware this field only supports `text_ad` or `spotlight` click uri types. If you are interested in this package supporting more click uri ad types, please let us know in this [Feature Request](https://github.com/fivetran/dbt_linkedin_source/issues/70).
+  - `stg_linkedin_ads__creative_history`
```
Comment (fivetran-avinash) [inline CHANGELOG.md:6]:
```suggestion
  - `stg_linkedin_ads__creative_history` (in the `dbt_linkedin_source` package)
```

**Review by fivetran-avinash — APPROVED** [review]
One small suggestion but otherwise LGTM!

File: packages.yml (hunk 1 above)
Comment (fivetran-joemarkiewicz) [inline packages.yml:6]:
```suggestion
  - package: fivetran/linkedin_source
    version: [">=0.10.0", "<0.11.0"]
```

**Pull request merged**